### PR TITLE
AZP/RELEASE: Update release ci for jucx stage

### PR DESCRIPTION
**Pull request by avildema**
# Why
Fix a problem with launch dockers for JUCX stage, for example https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=14454&view=logs&j=e29bbc24-f342-5915-29b0-97abf7400e12&t=dfce8edc-31a3-4b85-ae66-f22d64bb9d09

# How
Refactoring conditions for release pipeline

### CODE REVIEW

**Comment by yosefe** [comment]
bot:pipe:retest

File: buildlib/jucx/jucx-publish.yml [hunk 1]
```diff
@@ -5,8 +5,12 @@ parameters:
 
 jobs:
   - job: jucx_release
-    condition: eq(variables['Build.Reason'], 'ResourceTrigger')
-    container: centos7
+    condition: eq(stageDependencies.Check_Commit.Check.outputs['Commit.Launch'], 'Yes')
+    container: ubuntu20_cuda11_2
+    pool:
+      name: MLNX
+      demands:
```
Comment (yosefe) [inline buildlib/jucx/jucx-publish.yml:12]: AFAIU this is the actual fix?

Comment (avildema) [reply buildlib/jucx/jucx-publish.yml:12]: yes

File: buildlib/az-distro-release.yml [hunk 2]
```diff
@@ -112,3 +112,6 @@ jobs:
           addChangeLog: false
           assetUploadMode: replace
           assets: "./$(artifact_name)"
+      - task: PostBuildCleanup@3
+        displayName: 'Clean Agent Directories'
+        condition: always()
```
[the remark this answers is not in the document]
Comment (avildema) [reply buildlib/az-distro-release.yml]: removed

**Review by yosefe — APPROVED** [review]
pls squash

**Comment by petro-rudenko** [comment]
It does not launch JUCX spanshot publish: 
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=15486&view=logs&j=dc293d6f-ac09-5b94-81f8-18aba04876d1

```
Evaluating: eq(stageDependencies['Check_Commit']['Check']['outputs']['Commit.Launch'], 'Yes')
Expanded: eq(Null, 'Yes')
Result: False
```

We want snapshot to be published on every merge commit to master

**Pull request merged**